### ISSUE TITLE
[Functional Test] Socket timeout when pushing docs to LiteServ

### ISSUE THREAD

**Issue by sethrosetter** [couchbase/couchbase-lite-java-core#1576]
Using 1.4-30, Android Nexus 6P - Api 25, it looks like pushing docs to the Listener results in a socket timeout. 

Failing test - test_initial_push_replication[True] - http://uberjenkins.sc.couchbase.com/job/android-listener-client-sync-gateway/89/testReport/junit/testsuites.listener.shared.client_sg/test_replication/test_initial_push_replication_True_/

Full logcat - http://uberjenkins.sc.couchbase.com/job/android-listener-client-sync-gateway/89/artifact/results/logs/LiteServAndroid-test_initial_push_replication%5BTrue%5D-2017-01-30%2010%3A15%3A16.884896.txt

**Comment by sethrosetter** [comment]
Seeing a similar time when creating a database here - http://uberjenkins.sc.couchbase.com/job/listener-client-client/4/testReport/testsuites.listener.shared.client_client/test_peer_2_peer_sanity/test_peer_2_peer_sanity/

**Comment by hideki** [comment]
```python
data = self._sock.recv(self._rbufsize)
```
SocketTimeout error occurs at PyTest side. Can we increase Python's SocketTimeout value?

**Comment by hideki** [comment]
This round of functional test does not have any failure.
http://uberjenkins.sc.couchbase.com:8080/job/android-listener-client-sync-gateway/114/console

**Comment by hideki** [comment]
I executed `test_peer_2_peer_sanity` P2P with the real device (Nexus 6) 5 times continuously.

I observed `ConnectionError: ('Connection aborted.', error(54, 'Connection reset by peer'))` once during this test. However, I could observe `ping <ip address of device>` from the machine, that another peer and PyTest are running, returns `Request timeout`  intermittently. I believe this `ConnectionError` could be caused by office wifi network issue.
